contracts/core/Stake.sol

```diff
@@ -0,0 +1,417 @@
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../gateway/EIP20Interface.sol";
+import "./StakeInterface.sol";
+
+/**
+ * @title The Stake controct trocks deposits, logouts, slashings etc. on origin.
+ */
+contract Stake is StakeInterface {
+
+    /* Events */
+
+    /** Emitted whenever a new validator is successfully deposited. */
+    event NewDeposit(
+        address indexed validatorAddress,
+        uint256 indexed stake
+    );
+
+    /** Emitted whenever a meta-block increase is successfully registered. */
+    event HeightIncreased(uint256 indexed newHeight);
+
+    /* Structs */
+
+    /** A registered validator, */
+    struct Validator {
+        /**
+         * The depositor is the address that did the deposit for this
+         * validator. Only the depositor is allowed to logout the validator or
+         * withdraw the stake.
+         */
+        address depositor;
+
+        /**
+         * This is the address that the validator uses to sign votes on
+         * auxiliary.
+         */
+        address auxiliaryAddress;
+
+        /**
+         * The stake initially equals the deposit. It can increase due to
```

Comment:
This line-> `It can increase due to withdrawal or slashing.`
Will it decrease or increase ?

Reply:
🤦‍♂️ decrease, thanks, will fix